File: ohara-testing-util/src/test/scala/com/island/ohara/integration/TestOharaTestUtil.scala
package com.island.ohara.integration

import com.island.ohara.common.rule.MediumTest
import org.junit.Test
import org.scalatest.Matchers

class TestOharaTestUtil extends MediumTest with Matchers {

  private[this] def setEnv(key: String, value: String) = {
    val field = System.getenv().getClass.getDeclaredField("m")
    field.setAccessible(true)
    val map = field.get(System.getenv()).asInstanceOf[java.util.Map[java.lang.String, java.lang.String]]
    map.put(key, value)
  }

  @Test
  def testLocalMethod(): Unit = {
    setEnv("ohara.it.workers", "123")
    val util = OharaTestUtil.workers()
    try an[RuntimeException] should be thrownBy util.brokersConnProps finally util.close()
  }
}

File: ohara-testing-util/src/test/java/com/island/ohara/integration/TestOharaTestUtil.java
/*
 * Copyright 2019 is-land
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.island.ohara.integration;

import com.island.ohara.common.rule.MediumTest;
import java.lang.reflect.Field;
import java.util.Map;
import org.junit.Test;

public class TestOharaTestUtil extends MediumTest {

  @Test(expected = RuntimeException.class)
  public void testLocalMethod() {
    setEnv("ohara.it.workers", "123");
    try (OharaTestUtil util = OharaTestUtil.workers(1)) {
      util.brokersConnProps();
    }
  }

  @SuppressWarnings("unchecked")
  private void setEnv(String key, String value) {
    try {
      Field field = System.getenv().getClass().getDeclaredField("m");
      field.setAccessible(true);
      Map<String, String> map = (Map<String, String>) field.get(System.getenv());
      map.put(key, value);
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
}
